src/osemgrep/targeting/Git_path.mli

```diff
@@ -0,0 +1,47 @@
+(*
+   Abstract type for a file path within a git project.
+
+   This avoids issues of Unix-style vs. Windows-style paths. All
+   git paths use '/' as a separator.
+*)
+
+type t = private { string : string; components : string list }
```

Comment:
maybe explain what string is? I guess the original argument to of_string?

Reply:
maybe rename components to segments, to mimic the terminology in Fpath?
Also why not using Fpath which already provides most of the functions here (append, segments, /, etc.)?

Reply:
I thought "component" was the standard name because that's the terminology Gerd was using in [Ocamlnet](http://projects.camlcity.org/projects/dl/ocamlnet-4.1.9/doc/html-main/Neturl.html). Daniel uses "segment" in Fpath. https://en.wikipedia.org/wiki/Path_(computing) uses "component" and https://en.wikipedia.org/wiki/URL uses "segment". https://www.rfc-editor.org/rfc/rfc3986#section-3.3 uses "segment" while using "path component" to refer to the whole path in a URL. Segment it is, then.